iOS-26 | Добавлено кэширование сообщений

mightyK1ngRichard opened the pull request.

mightyK1ngRichard committed:
iOS-26: Добавил кэширование сообщений

mightyK1ngRichard committed:
iOS-26: Добавил комментарии к коду

mightyK1ngRichard committed:
iOS-26: Мелкие фиксы

mightyK1ngRichard committed:
iOS-26: Добавлен запрос истории чата из профиля

mightyK1ngRichard committed:
iOS-27: Добавлено кэширование фоток. Обновление моделей

mightyK1ngRichard committed:
iOS-27: Удалил тестовый экран

itissep commented on CakesHub/CakesHub/Services/Firebase/Entities/FBProductModel/FBProductModel.swift:
```diff
@@ -97,15 +97,23 @@ extension FBProductModel {
-            [.day],
-            from: establishmentDate.toDate,
+        guard let productDate = establishmentDate.dateRedescription else {
+            return false
+        }
+        let componentsDif = Calendar.current.dateComponents(
+            [.year, .month, .day],
+            from: productDate,
             to: Date.now
-        ).day else { return false }
+        )
+        guard componentsDif.month == 0 && componentsDif.year == 0 else {
+            return false
+        }
+
+        // Получаем разницу нынешней даты и даты создания объявления
+        guard let difDay = componentsDif.day else { return false }
```
вот это бы стоило иметь где-нибудь в хелперах или расширениях

mightyK1ngRichard committed:
iOS-28: Создал экран создания отзыва

mightyK1ngRichard committed:
iOS-28: Добавил логику для экрана отзыва

mightyK1ngRichard committed:
iOS-28: Доработка экрана отзыва и моделей

1. Добавил обновление информации о товаре при оставке отзыва
2. Поправил расчёты логики рейтинга товара

mightyK1ngRichard committed:
iOS-28: Убрал анимацию при обновлении категорий

mightyK1ngRichard committed:
iOS-29: Добавил локальные пуши с задержкой

mightyK1ngRichard committed:
iOS-30: Обновил шаблон

mightyK1ngRichard committed:
iOS-30: Добавлена карта

mightyK1ngRichard committed:
iOS-30: minor

mightyK1ngRichard committed:
iOS-30: Minor

mightyK1ngRichard committed:
iOS-30: fix bug

mightyK1ngRichard committed:
iOS-31: Добавил архитектуру для экрана насроек и логику выхода

mightyK1ngRichard committed:
iOS-31: Доработка AuthView под новый подход

mightyK1ngRichard committed:
iOS-31: minor

mightyK1ngRichard committed:
Merge pull request #43 from iOS-vk-education/feature/permyakov/iOS-31_logout

iOS-31 | Добавлена архитектура для экрана настроек. Внедрена логика выхода их аккаунта

mightyK1ngRichard committed:
Merge pull request #42 from iOS-vk-education/feature/permyakov/iOS-30_map_view

iOS-30 | Добавлена карта

mightyK1ngRichard committed:
Merge pull request #41 from iOS-vk-education/feature/permyakov/iOS-29_add_local_push_notifications

iOS-29 | Добавлены локальные пуши

mightyK1ngRichard committed:
Merge pull request #40 from iOS-vk-education/feature/permyakov/iOS-28_add_feedback_screen

iOS-28 | Добавлен экран фидбэка о товаре

mightyK1ngRichard committed:
Merge pull request #39 from iOS-vk-education/feature/permyakov/iOS-27_image_cache

iOS-27 | Добавил кэширование фотографий в FileManager
